src/shared/dissect-image.c

```diff
@@ -4672,6 +4674,29 @@ int verity_dissect_and_mount(
         return 0;
 }
 
+int verity_sig_derive_uuid(const void *root_hash, size_t root_hash_size, const sd_id128_t *partition_guid, sd_id128_t *ret_uuid) {
+        union {
+                uint8_t md[SHA256_DIGEST_SIZE];
+                sd_id128_t id;
+        } result;
+
+        assert(root_hash);
+        assert(root_hash_size > 0);
+        assert(partition_guid);
+        assert(ret_uuid);
+
+        /* As per DPS, derive the verity signature UUID from the root hash and partition GUID. */
+
+        hmac_sha256(root_hash, root_hash_size, partition_guid->bytes, sizeof(partition_guid->bytes), result.md);
```

Comment:
hmm, hmac doesn't really make sense i guess, after all. hmac is good if you have a "secret" and some "data" and you want to securely combine them in a hash. for the /var/ case we hence use HMAC with the machine-id as secret (because it probably shouldn#t leak too obviously outside), similar in repart: the "seed" we want to keep secret (since in many cases it is the machine ID), hence we use it as a secret.

hence i don't think we should use hmac here.

hence, do "SHA256(CONCAT(partition-type-uuid, roothash))` here.